Add Attributes to Relation

baldurrensch opened the pull request:
This came out of discussions regarding https://github.com/TheFootballSocialClub/FSCHateoasBundle/pull/38. We decided that it would provide more flexibility if the relation had an array of attributes, one of them being `isTemplated`. 

This fixes #38

Commit:
Added attributes to relation

Commit:
Added attributes to link

Commit:
Added isTemplated to relation attributes

adrienbrault commented on Model/Link.php:
```diff
@@ -9,6 +9,7 @@ class Link
 {
     private $rel;
     private $href;
+    private $relationAttributes;
```
I'd just name this attributes

baldurrensch replied:
The problem is that we have attributes on the relation, options and parameters on the route. Get's confusing, doesn't it?

adrienbrault replied:
parameters are on the route, options on the route, attributes on the relation
I don't think that it's really confusing.

adrienbrault commented on Serializer/LinkSerializationHelper.php:
```diff
@@ -23,6 +23,11 @@ public function addLinksToXMLSerialization(array $links, XmlSerializationVisitor
             $entryNode->setAttribute('rel', $link->getRel());
             $entryNode->setAttribute('href', $link->getHref());
+
+            $relationAttributes = $link->getRelationAttributes();
```
You should loop through the relationAttributes and set every one of them on the node

```
foreach ($link->getRelationAttributes() as $attributeName => $attributeValue) {
    $entryNode->setAttribute($attributeName, $attributeValue);
}
```

adrienbrault commented on Serializer/LinkSerializationHelper.php:
```diff
@@ -42,6 +47,11 @@ public function createGenericLinksData(array $links, GenericSerializationVisitor
 
+            $relationAttributes = $link->getRelationAttributes();
+            if (isset($relationAttributes['isTemplated'])) {
+                $serializedLink['templated'] = $relationAttributes['isTemplated'] ? 'true' : 'false';
```
```
$serializedLink = array_merge($serializedLink, $link->getRelationAttributes());
```

Commit:
Small fixes

- changed relationAttribute to attribute in Link Model
- added all attributes to serialization

adrienbrault commented on Factory/AbstractLinkFactory.php:
```diff
@@ -18,11 +18,12 @@ public function __construct(RelationUrlGenerator $relationUrlGenerator, $forceAb
     }
 
-    public static function createLink($rel, $href)
+    public static function createLink($rel, $href, $relationAttributes = null)
```
this should be named `$attributes`

Commit:
Change all relationAttribute to attribute

Commit:
Update Readme

adrienbrault commented:
Thanks!
